src/lib/core/datetime/simple-date.spec.ts

```diff
@@ -9,4 +9,32 @@ describe('SimpleDate', () => {
   it('can be converted to native Date', () => {
     expect(new SimpleDate(2017, 0, 1).toNativeDate()).toEqual(new Date(2017, 0, 1));
   });
+
+  it('handles month and date overflow', () => {
+    expect(new SimpleDate(2017, 12, 32)).toEqual(new SimpleDate(2018, 1, 1));
+  });
+
+  it('handles month and date underflow', () => {
+    expect(new SimpleDate(2017, -1, 0)).toEqual(new SimpleDate(2016, 10, 30));
```

Comment:
The months are 0-indexed to match native JS Dates, so 10 = November. I considered using 1-indexed because it make more sense, but I thought it was better to just stay consistent with what people already know. I'll add a note about the 0-indexing to the SimpleDate docs.